src/main/scala/us/diempham/taskman/web/services/TaskService.scala

```diff
@@ -1,23 +1,47 @@
 package us.diempham.taskman.web.services
 
+import java.util.UUID
+import cats.implicits._
 import cats.effect.IO
 import org.http4s._
 import org.http4s.dsl.io._
 import us.diempham.taskman.application.TaskStorage
-import us.diempham.taskman.application.TaskStorage.{TaskId, Task}
-import us.diempham.taskman.database.InMemoryDatabase
-import us.diempham.taskman.web.services.domain.CreateTaskRequest
+import us.diempham.taskman.application.TaskStorage.{IsCompleted, Task, TaskId}
+import us.diempham.taskman.application.UserStorage.{User, UserId}
+import us.diempham.taskman.database.{InMemoryDatabase, InMemoryTaskDatabase}
+import us.diempham.taskman.web.services.domain.{CreateTaskRequest, taskToTaskResponse}
+import us.diempham.taskman.web.services.extractor.TaskIdExtractor
 
-class TaskService(taskStorage: InMemoryDatabase[TaskId, Task]) {
+class TaskService(taskStorage: InMemoryTaskDatabase, userStorage: InMemoryDatabase[UserId, User]) {
+  val TASKS = "tasks"
 
- val service = HttpService[IO] {
-   case request @ POST -> Root / "tasks" =>
-     for {
-       task <- request.as[CreateTaskRequest]
-       insertTask = taskStorage.create(task.taskId, TaskStorage.Task(task.taskId, task.userId, task.title, task.description, task.isCompleted, task.createdOn))
-       response <- Ok(())
-     } yield response
+  val service = HttpService[IO] {
+    case request @ POST -> Root / TASKS =>
+      for {
+        task <- request.as[CreateTaskRequest]
+        response <- maybeInsertTask(task) match {
+            case None => NotFound("User doesn't exist!")
+            case Some(task) => Ok(taskToTaskResponse(task))
+          }
+        } yield response
 
 
- }
+    case PUT -> Root / TASKS / TaskIdExtractor(taskId) / "completed" =>
```

Comment:
What about marking a task as incomplete? Just in case I accidentally mark something complete that I didn't mean to.